### PR TITLE
Build: switch to webpack

### PR DESCRIPTION
**Pull request by g-plane**
Switching to webpack, it allows us to use ES6 `import` in source file.

### CODE REVIEW

File: js/app/.eslintrc.yml [hunk 1]
```diff
@@ -13,6 +13,7 @@ parserOptions:
     ecmaVersion: 5
     ecmaFeatures:
         jsx: true
+    sourceType: module
```
Comment (aladdin-add) [inline js/app/.eslintrc.yml]: I don't think we can use module here, given ecmaVersion is setting to 5.

File: webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const path = require("path");
+
+module.exports = {
+    mode: "production",
+    entry: "./js/app/parser/index.js",
+    output: {
+        filename: "index.built.js",
+        path: path.resolve("./js/app/parser")
```
Comment (aladdin-add) [inline webpack.config.js]: can you use `path.join()` here,  I remember we encountered a similar issue in eslint repo somehow.

File: js/app/.eslintrc.yml (hunk 1 above)
Comment (g-plane) [reply js/app/.eslintrc.yml]: So should we revert it to use CommonJS instead of ES Modules?

Comment (aladdin-add) [reply js/app/.eslintrc.yml]: I guess so.…（⊙＿⊙；）…
 another way would be adding something like babel, but it needs to be discussed more.

File: webpack.config.js (hunk 2 above)
Comment (g-plane) [reply webpack.config.js]: The `path` field must be absolute path, while using `path.join` won't return that.

Comment (kaicataldo) [reply webpack.config.js]: I think this is fine, though I do think it's common to use `__dirname`. i.e. `path.resolve(__dirname, "js/app/parser")`.

File: js/app/.eslintrc.yml (hunk 1 above)
Comment (kaicataldo) [reply js/app/.eslintrc.yml]: I agree - I don't think we should necessarily update to use ES modules here. That seems like a bigger discussion around if we want to use other ES2015+ features or not. I've been away for a bit (so please correce me if I'm wrong!), but I don't think we actually have a real automated build step at the moment.

**Comment by g-plane** [comment]
I'm closing now since I think current build steps work fine.

**Pull request closed without merging**